src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -62,7 +77,7 @@ protected Void doInBackground() throws Exception {
 			HttpMessage msg = this.httpMessage.cloneRequest();
 
 			try {
-				msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				msg.getRequestHeader().setCookieParams(cookies);

```

Comment:
Re-setting all the cookies prevents the regeneration of session (form) tokens (i.e. the same session is used in all requests and the form token ends up being the same).

This changes (both re-setting the cookies as well removing them) could be done, just once, before the for loop, we just need to clone the request, modify it and then use that message (i.e. assign the modified message to instance variable httpMessage).
What do you think?